operator/api/v1alpha1/observatorium_types.go

```diff
@@ -158,6 +158,8 @@ type APISpec struct {
 	Image string `json:"image,omitempty"`
 	// Version describes the version of API to use.
 	Version string `json:"version,omitempty"`
+	// TLS Secret Configuration
+	Secret string `json:"secret,omitempty"`
```

Comment:
This should probably be named `TLSSecretName` or something more descriptive. Also, I would rather propose making this a struct called `TLS` with a sub field called `SecretName`, `CAField`, `CertificateField` etc otherwise it is super hard to communicate to the user how to structure their secret correctly. It's much easier to ask the user to tell us how to structure it IMO

Reply:
Done